File: src/main/java/wraith/fabricaeexnihilo/util/VanillaWoodDefinitions.java
package wraith.fabricaeexnihilo.util;

import net.minecraft.util.Identifier;
import wraith.fabricaeexnihilo.FabricaeExNihilo;
import wraith.fabricaeexnihilo.api.WoodDefinition;

public enum VanillaWoodDefinitions implements WoodDefinition {
    OAK("oak"),
    SPRUCE("spruce"),
    BIRCH("birch"),
    JUNGLE("jungle"),
    ACACIA("acacia"),
    DARK_OAK("dark_oak");
    
    private final String text;
    
    VanillaWoodDefinitions(String text) {
        this.text = text;
    }
    
    public Identifier getTexturePlanks() {
        return new Identifier("block/" + text + "_planks");
    }
    
    public Identifier getTextureLog() {
        return new Identifier("block/" + text + "_log");
    }
    
    public Identifier getTextureLogTop() {
        return new Identifier("block/" + text + "_log_top");
    }
    
    public Identifier getLeafID() {
        return new Identifier(text + "_leaves");
    }
    
    public Identifier getLogID() {
        return new Identifier(text + "_log");
    }
    
    public Identifier getPlanksID() {
        return new Identifier(text + "_planks");
    }
    
    public Identifier getSlabID() {
        return new Identifier(text + "_slab");
    }
    
    public Identifier getSaplingID() {
        return new Identifier(text + "_sapling");
    }
    
    public Identifier getSeedID() {
        return FabricaeExNihilo.id("seed_" + text);
    }
    
}

File: src/main/kotlin/exnihilofabrico/util/VanillaWoodDefinitions.kt
package exnihilofabrico.util

import exnihilofabrico.api.IWoodDefinition
import exnihilofabrico.id
import net.minecraft.util.Identifier

enum class VanillaWoodDefinitions(val text: String): IWoodDefinition {
    OAK("oak"),
    SPRUCE("spruce"),
    BIRCH("birch"),
    JUNGLE("jungle"),
    ACACIA("acacia"),
    DARK_OAK("dark_oak");

    override fun getTexturePlanks() = Identifier("block/${text}_planks")
    override fun getTextureLog() = Identifier("block/${text}_log")
    override fun getTextureLogTop() = Identifier("block/${text}_log_top")

    override fun getLeafID() = Identifier("${text}_leaves")
    override fun getLogID() = Identifier("${text}_log")
    override fun getPlanksID() = Identifier("${text}_planks")
    override fun getSlabID() = Identifier("${text}_slab")
    override fun getSaplingID() = Identifier("${text}_sapling")

    override fun getSeedID() = id("seed_$text")
}